wicket-core/src/main/java/org/apache/wicket/markup/head/JavaScriptHeaderItem.java

```diff
@@ -201,7 +202,13 @@ public static JavaScriptReferenceHeaderItem forReference(ResourceReference refer
 	public static JavaScriptReferenceHeaderItem forReference(ResourceReference reference,
 		PageParameters pageParameters, String id, boolean defer, String charset)
 	{
-		return new JavaScriptReferenceHeaderItem(reference, pageParameters, id, defer, charset, null);
+                final JavaScriptReferenceHeaderItem javaScriptReferenceHeaderItem = new JavaScriptReferenceHeaderItem(reference, pageParameters, id, defer, charset, null);
+                if(reference instanceof IntegrityAttributed) {
```

Comment:
Does a reference really need to be able to decide about integrity attributes? It can't change charset or defer either, why are integrity attributes treated differently?